### PR TITLE
[@mantine/form] use-form: add submitting and submitError + methods

### PR DESCRIPTION
**Pull request by philipodev**
as per #4011 

### use-form

* Added `submitting`, `submitError` fields
* Added `setSubmitting`, `setSubmitError` and `resetSubmitError` methods

### CODE REVIEW

**Review by treemmett — APPROVED** [review]
This looks great! Hopefully this gets implemented soon.

File: src/mantine-form/src/use-form.ts [hunk 1]
```diff
@@ -50,6 +54,8 @@ export function useForm<
   const [dirty, setDirty] = useState(initialDirty);
   const [values, _setValues] = useState(initialValues);
   const [errors, _setErrors] = useState(filterErrors(initialErrors));
+  const [submitting, _setSubmitting] = useState(false);
+  const [submitError, _setSubmitError] = useState<SubmitError>(null);
```
Comment (treemmett) [inline src/mantine-form/src/use-form.ts:58]: I think these should get reset in the [`reset` callback](https://github.com/mantinedev/mantine/pull/4020/files#diff-885f0cbfb62b24bda2ed7145a69597b51a67fabc68b78fde5b672962ed97559eR79)

**Review by treemmett — CHANGES_REQUESTED** [review]

**Comment by laugri** [comment]
Hello, will this be implemented at some point ?

**Comment by ldalves** [comment]
We are currently migrating a huge project from `React Hook Form` to `mantine/form` and we are very interested in this feature. Any estimates when this might be released?

**Comment by stevewillard** [comment]
Any chance this could make it into the next release?

**Comment by ldalves** [comment]
why close? 😢  was created a separate PR and merged it or something like that?

**Comment by SergeyNedayvodaBit** [comment]
hi, is this functionality still in the plans?

**Comment by geoguide** [comment]
Weird silent close

**Comment by Akronae** [comment]
I was really hoping for this feature just to show a loading spinner on my submit button 😭

**Pull request closed without merging**